src/panels/lovelace/header-footer/hui-graph-header-footer.ts

```diff
@@ -125,87 +129,89 @@ export class HuiGraphHeaderFooter
+  private _subscribeHistoryTimeWindow() {
+    if (!isComponentLoaded(this.hass!, "history") || this._subscribed) {
+      return;
     }
+    this._subscribed = subscribeHistoryStatesTimeWindow(
+      this.hass!,
+      (combinedHistory) => {
+        if (!this._subscribed) {
+          // Message came in before we had a chance to unload
+          return;
+        }
+        this._coordinates =
+          coordinatesMinimalResponseCompressedState(
+            combinedHistory[this._config!.entity],
+            this._config!.hours_to_show!,
+            500,
+            this._config!.detail!,
+            this._config!.limits
+          ) || [];
+      },
+      this._config!.hours_to_show!,
+      [this._config!.entity]
+    ).catch((err) => {
+      this._subscribed = undefined;
+      this._error = err;
+    });
+    this._setRedrawTimer();
   }
 
-  private async _getCoordinates(): Promise<void> {
-    this._fetching = true;
-    const endTime = new Date();
-    const startTime =
-      !this._date || !this._stateHistory?.length
-        ? new Date(
-            new Date().setHours(
-              endTime.getHours() - this._config!.hours_to_show!
-            )
-          )
-        : this._date;
-
-    if (this._stateHistory!.length) {
-      const inHoursToShow: HassEntity[] = [];
-      const outHoursToShow: HassEntity[] = [];
-      // Split into inside and outside of "hours to show".
-      this._stateHistory!.forEach((entity) =>
-        (endTime.getTime() - new Date(entity.last_changed).getTime() <=
-        this._config!.hours_to_show! * HOUR
-          ? inHoursToShow
-          : outHoursToShow
-        ).push(entity)
-      );
-
-      if (outHoursToShow.length) {
-        // If we have values that are now outside of "hours to show", re-add the last entry. This could e.g. be
-        // the "initial state" from the history backend. Without it, it would look like there is no history data
-        // at the start at all in the database = graph would start suddenly instead of on the left side of the card.
-        inHoursToShow.push(outHoursToShow[outHoursToShow.length - 1]);
-      }
-      this._stateHistory = inHoursToShow;
+  private _redrawGraph() {
+    if (this._coordinates) {
+      this._coordinates = [...this._coordinates];
     }
+  }
 
-    const stateHistory = await fetchRecent(
-      this.hass!,
-      this._config!.entity,
-      startTime,
-      endTime,
-      Boolean(this._stateHistory!.length)
+  private _setRedrawTimer() {
+    // redraw the graph every minute to update the time axis
+    clearInterval(this._interval);
+    this._interval = window.setInterval(
+      () => this._redrawGraph(),
+      this._config!.hours_to_show! > 24 ? HOUR : MINUTE
     );
+  }
 
-    if (stateHistory.length && stateHistory[0].length) {
-      this._stateHistory!.push(...stateHistory[0]);
+  private _unsubscribeHistoryTimeWindow() {
+    if (!this._subscribed) {
+      return;
     }
+    clearInterval(this._interval);
```

Comment:
We need to move this above the if-check. If the subscribe call failed, we still set the interval.

Reply:
adjusted in 265ca40df1525f97c6819a01cd2865a75da73dd4